katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,28 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  POSTGRES_DIR = '/var/lib/pgsql'
+
+  begin
+    available_space = `df -k --total --exclude-type=tmpfs --output=avail`.split("\n")[-1].to_i
```

Comment:
I think this should get the disk space available on `/var/opt/rh/rh-postgresql10/lib/pgsql/data` (or the closest path that does exist). Total disk space isn't quite correct.
```ruby
new_postgres_dir = '/var/opt/rh/rh-postgresql10/lib/pgsql/data'
until File.directory?(new_postgres_dir)
  new_postgres_dir = File.dirname(new_postgres_dir)
end
available_space = `df -k --output=avail "#{new_postgres_dir}"`.split("\n").last.to_i
```

In `katello/hooks/pre_validations/30-mongo_storage_engine.rb` there's similar code so it may make sense to create a helper for this at some point :)